index.bs

```diff
@@ -257,28 +264,111 @@ spec:css-syntax-3;
   This document depends on the Infra Standard for a number of foundational concepts used in its
   algorithms and prose [[!INFRA]].
 
+
+  ### Infrastructure Algorithms ### {#sctn-infra-algorithms}
+
+  <!-- NOTE: '<h4>' maps to the same heading level as three hashs '###', thus using '<h5>' for the next section -->
+  <h5 id="algorithm-same-origin-with-ancestors" algorithm>Same-Origin with its Ancestors</h5>
+
   An [=environment settings object=] (|settings|) is <dfn noexport>same-origin with its
   ancestors</dfn> if the following algorithm returns `true`:
 
-  1.  If |settings| has no [=environment settings object/responsible document=],
-      return `false`.
+  <ol class="algorithm">
+    1.  If |settings| has no [=environment settings object/responsible document=],
+        return `false`.
 
-  2.  Let |document| be |settings|' [=environment settings object/responsible document=].
+    2.  Let |document| be |settings|' [=environment settings object/responsible document=].
 
-  3.  If |document| has no [=document/browsing context=], return `false`.
+    3.  If |document| has no [=document/browsing context=], return `false`.
 
-  4.  Let |origin| be |settings|' [=environment settings object/origin=].
+    4.  Let |origin| be |settings|' [=environment settings object/origin=].
 
-  5.  Let |current| be |document|'s [=document/browsing context=].
+    5.  Let |current| be |document|'s [=document/browsing context=].
 
-  6.  While |current| has a [=parent browsing context=]:
+    6.  While |current| has a [=parent browsing context=]:
 
-      1.  Set |current| to |current|'s [=parent browsing context=].
+        1.  Set |current| to |current|'s [=parent browsing context=].
+
+        2.  If |current|'s [=active document=]'s [=origin=] is not [=same origin=] with |origin|,
+            return `false`.
+
+    7.  Return `true`.
+  </ol>
 
-      2.  If |current|'s [=active document=]'s [=origin=] is not [=same origin=] with |origin|,
-          return `false`.
+  ### Registry: Credential Type to Options Member Identifier Mappings ### {#sctn-registry-cred-type-to-options-member-identifier}
+
+  This registry maps [=credential types=] (i.e., {{Credential/[[type]]}} values) to the
+  [=dictionary member=] [=identifier=] used in
+  {{CredentialCreationOptions}} and {{CredentialRequestOptions}} (i.e., "options dictionaries")
+  by their specifications.
+
+  Note: This registry is used by the [=relevant credential interface objects=] algorithm.
+
+
+  <table class="data">
+    <thead>
+      <tr>
+        <th>Credential Type<br>
```

Comment:
It looks like we refer to this from algorithms' steps, so it should be 
```suggestion
        <th><dfn for="Credential Type Registry">Credential Type</dfn><br>
```
or similar so we can link instead of just naming the column.